webapp/src/ts/services/db-sync.service.ts

```diff
@@ -166,31 +168,76 @@ export class DBSyncService {
       telemetryEntry.recordSuccess(result);
     } catch (err) {
       telemetryEntry.recordFailure(err, this.knownOnlineState);
-      console.error(`Error replicating from remote server`, err);
+      console.error('Error replicating from remote server', err);
       return err;
     }
   }
 
+  private getCurrentSeq(): Promise<number> {
+    return this.dbService.get().info().then(info => info.update_seq);
+  }
 
-  private getCurrentSeq() {
-    return this.dbService.get().info().then(info => info.update_seq + '');
+  private getLastReplicatedSeq(): number {
+    return Number(window.localStorage.getItem(LAST_REPLICATED_SEQ_KEY)) || 0;
   }
 
-  private getLastReplicatedSeq() {
-    return window.localStorage.getItem(LAST_REPLICATED_SEQ_KEY);
+  private setLastReplicatedSeq(sequence: number) {
+    if (!sequence) {
+      return;
+    }
+
+    const lastReplicatedSeq = this.getLastReplicatedSeq();
+    if (lastReplicatedSeq > sequence) {
+      return;
+    }
```

Comment:
I don't think this block is necessary. If `lastReplicatedSeq` is greater than `sequence` then writing it to local storage will still mean replication is retried because they're not equal. So the code can be simplified by removing these 4 lines.

It's also more correct to update it because technically we know the replication is complete up to `sequence` so we should store that.